annotation/annotation/tasks/services.py

```diff
@@ -62,6 +61,37 @@
     ValidationSchema,
 )
 
+
+class LRU:
+    def __init__(self, capacity):
+        self.capacity = capacity
+        self.counter = 0
+        self.cache = {}
+        self.lru = {}
+
+    def __contains__(self, item):
+        if item in self.cache:
+            return True
+        else:
+            return False
+
+    def __getitem__(self, key):
+        if key in self.cache:
+            self.lru[key] = self.counter
+            self.counter += 1
```

Comment:
The counter can be incremented only up to the capacity; you don't need to go beyond that. Use modulo for this.

Reply:
Smth like priority queue